### DISCUSSION TITLE
CHILD_SA not beeing established if the other side deletes it

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#1877] [General]
Hi,

I saw the following behaviour on my strongswan server:
charon[13245]: 02[IKE] received DELETE for ESP CHILD_SA with SPI 6848609c
charon[13245]: 02[IKE] closing CHILD_SA conn-net{1} with SPIs c7b40c2a_i (0 bytes) 6848609c_o (0 bytes) and TS x.x.x.x === y.y.y.y
charon[13245]: 02[IKE] closing CHILD_SA conn-net{1} with SPIs c7b40c2a_i (0 bytes) 6848609c_o (0 bytes) and TS x.x.x.x === y.y.y.y
charon[13245]: 02[IKE] sending DELETE for ESP CHILD_SA with SPI c7b40c2a
charon[13245]: 02[IKE] CHILD_SA closed

but no further attempt to re-establish the CHILD_SA. Further communication over the tunnel was not possible and did not trigger a renegotiation of the CHILD_SA.

Any idea how I can set the child up for permanent active tunnel?

I have DPD configued. DPD INFORMS are sent and replied. No problem here.

**Comment by tobiasbrunner** [comment]
Use trap policies so the SA is automatically created on matching traffic and to avoid traffic leaks. You could also use `close_action = restart` to react to such deletes (there is a potential for loops, depending on what the other end is doing exactly).

**Reply by shmeleviv** [reply]
Hello!!

@tobiasbrunner  could you please explain possible reason of traffic leaks? Several days ago i observed the same behavior:
1) initially, child_sa works fine
2) after received DELETE from remote side, CHILD_SA stuck in DOWN state and not deleted
3) communication over the tunnel was not possible and did not trigger a renegotiation of the CHILD_SA (traffic is initiated from stronswan side)

**Reply by shmeleviv** [reply]
UPD
start_action = trap is already set. Issue is still observed

**Reply by tobiasbrunner** [reply]
> 2\. after received DELETE from remote side, CHILD_SA stuck in DOWN state and not deleted

What does that mean? What DOWN state?

> 3\. communication over the tunnel was not possible and did not trigger a renegotiation of the CHILD_SA

Why? What exactly is happening?

You have to realize that I can't help you at all if you don't provide information such as logs etc.
